### DISCUSSION TITLE
fcitx5 does not work with VNC clients

### DISCUSSION THREAD

**Discussion by burbilog** [fcitx/fcitx5#602] [Q&A]
Kubuntu 24.04.1 LTS. Fcitx5 does not switch keyboards for any VNC client. I tried:

1. Remmina. Does not budge. Does not change keyboard at all.
2. RealVnc for linux from their site. The same.
3. xtightvncviewer. The same.
4. tigervnc-viewer. The same.
5. tightvnc-java. The same.

Even tried novnc from snap, logging through the browser, it does not switch the keyboard either! It can switch keyboards in address entry bar, but can't switch address when novnc tab is in focus.

Right now the only way to type non-english text remotely is Widnows Virtualbox machine with any VNC viewer...

But it works in general on local machine -- browser, libreoffice, konsole, etc. 

fcitx5-diagnose output: https://gist.github.com/burbilog/c7a02bf5b7fb7904dc2b656026d64469

**Comment by wengxt** [comment]
Can you explain what you're trying to do?

Are you running fcitx on your host, or the vnc remote machine?

It seems like you're trying to use fcitx to type into remote machine, which won't work.

You need to run a fcitx on the remote side to use it.

**Reply by burbilog** [reply]
I'm running fcitx on local machine and trying to type in RDP window. This is pretty expected behavior.

A few system updates later fcitx is now working in RDP clients.
